rsciio/quantumdetector/_api.py

```diff
@@ -344,6 +355,10 @@ def load_mib_data(
         data = data.rechunk(chunks)
 
     if return_headers:
+        if distributed:
```

Comment:
You can still return the header by just setting the `key="header"` for a second `memmap_distributed` call.  It will add some time onto the saving of the dataset as the entire dataset might get loaded into ram with most of it thrown away. 

Really what we should do is add things to a `to_store` context manager and then call:
https://github.com/hyperspy/rosettasciio/blob/31bd677cc4c02c5787ae9b61250bc1431f352cba/rsciio/hspy/_api.py#L111

Only once.  That will merge taskgraphs as necessary and might reduce the time for saving certain signals.  I've thought about it for things like saving lazy markers of possibly creating a `hs.save()` function for handling mulitple signals if you wanted to save multiple parts of some anaylsis efficently.  This is a fairly abstract/higher level concept so maybe it would be seledomly used.

Reply:
Yes, this will most likely needed to be done at some point! I opened #269 to track it / add more usecases.